apps/parsley/src/hooks/useSections/index.ts

```diff
@@ -108,32 +116,37 @@ export const useSections = ({
     [],
   );
 
+  /**
+   * This function will update the current section state and returns the next section state where
+   * all sections that contain the lineNumber values(s) are open. If the next state is the same as the current state,
+   * the function will return true and false otherwise.
+   * @param param0 is an object with a lineNumber key(s).
+   * @param param0.lineNumber is a number or an array of numbers that represent raw log line numbers.
+   * @returns true if the sectionState was updated and false otherwise
+   */
+  const openSectionContainingLineNumber = ({
+    lineNumber,
+  }: {
+    lineNumber: number | number[];
+  }): boolean => {
+    if (!sectionData || !sectionState) {
+      return false;
+    }
+    const nextState = openSectionContainingLineNumberHelper({
+      lineNumber,
+      sectionData,
+      sectionState,
+    });
+    setSectionState(nextState);
+    return nextState !== sectionState;
```

Comment:
I like it this way mainly because of code style/avoid an extra condition. This doesn't have a perf implication since the equality check is reference based.